### PR TITLE
Clean up product management resources and move design to engineering

### PR DESCRIPTION
**Pull request by malomarrec**
- Clean up product management resources / process and deprecate old content
- Serendipity "Given I broke everything, let me break more things" update: Design is now under the umbrella engineering team, not under product

### CODE REVIEW

**Comment by malomarrec** [comment]
@sqs this started as a very small cleanup of product pages, but I broke so many links that I took the opportunity to move the design folder as well and fix everything in one go.
Tagging you as it may interfere with your last priority [here](https://sourcegraph.slack.com/archives/C02KZ941RNX/p1656344683826349), in which case let me know and I'll close this PR.

File: content/departments/engineering/product/process/index.md [hunk 1]
```diff
@@ -4,43 +4,11 @@ This page contains information that is relevant for how to do well at your job a
 
 ## Product process
```
Comment (malomarrec) [inline content/departments/engineering/product/process/index.md:5]: @Joelkw @ryankscott the most meaningful changes are in this file, the rest is links being fixed and minor tweaks.

**Comment by malomarrec** [comment]
@sourcegraph/product  I'm gonna leave this open one more day, then merge it unless I hear back :)

**Review by ryankscott — APPROVED** [review]
LGTM

**Pull request merged**